### PR TITLE
feature/math_library_wrap_around

### PR DESCRIPTION
**Pull request by VectorGamez**
Add base wrap function

Add math blueprint library class
Use base wrap function in array wrap

### CODE REVIEW

File: Source/GameBaseFramework/Public/BlueprintLibraries/GBFMathBlueprintLibrary.h [hunk 1]
```diff
@@ -0,0 +1,18 @@
...
+
+#include "GBFMathBlueprintLibrary.generated.h"
+
+UCLASS()
+class GAMEBASEFRAMEWORK_API UGBFMathBlueprintLibrary : public UBlueprintFunctionLibrary
+{
+    GENERATED_BODY()
+
+public:
+    static int GetWrappedIndex( const int index, const int size )
```
Comment (TheEmidee) [inline Source/GameBaseFramework/Public/BlueprintLibraries/GBFMathBlueprintLibrary.h:14]: UFUNCTION( BlueprintPure )

if not possible, UFUNCTION( BlueprintCallable )

File: Source/GameBaseFramework/Public/BlueprintLibraries/GBFMathBlueprintLibrary.h [hunk 2]
```diff
@@ -0,0 +1,18 @@
...
+    {
+        return ( ( index % size ) + size ) % size;
```
Comment (TheEmidee) [inline Source/GameBaseFramework/Public/BlueprintLibraries/GBFMathBlueprintLibrary.h:16]: move the implementation in the cpp file

File: Source/GameBaseFramework/Public/BlueprintLibraries/GBFMathBlueprintLibrary.h (hunk 1 above)
Comment (VectorGamez) [reply Source/GameBaseFramework/Public/BlueprintLibraries/GBFMathBlueprintLibrary.h]: Done, BlueprintPure now.

File: Source/GameBaseFramework/Public/BlueprintLibraries/GBFMathBlueprintLibrary.h (hunk 2 above)
Comment (VectorGamez) [reply Source/GameBaseFramework/Public/BlueprintLibraries/GBFMathBlueprintLibrary.h]: Done.

**Pull request merged**